src/webapp/forms.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+from typing import List
+
+from flask_wtf import FlaskForm
+from wtforms import BooleanField, SelectField, SelectMultipleField, StringField, \
+    TimeField, HiddenField
+from wtforms.ext.dateutil.fields import DateField
+from wtforms.validators import InputRequired
+
+from webapp.common import gen_id
+
+
+class GenerateDowntimeForm(FlaskForm):
+    scheduled = BooleanField("Scheduled", None)  # default="checked" does not do anything
+    severity = SelectField("Severity", [InputRequired()], choices=[
+        ("Outage", "Outage"),
+        ("Severe", "Severe"),
+        ("Intermittent Outage", "Intermittent Outage"),
+        ("No Significant Outage Expected", "No Significant Outage Expected")
+    ])
+    description = StringField("Description", [InputRequired()])
+    start_date = DateField("Start Date", [InputRequired()])
+    start_time = TimeField("Start Time", [InputRequired()])
+    end_date = DateField("End Date", [InputRequired()])
+    end_time = TimeField("End Time", [InputRequired()])
+    services = SelectMultipleField("Services (select one or more)", [InputRequired()], choices=[])
```

Comment:
Can we give these fields tooltips?

Reply:
That looks like it would take a lot of code.